lib/base.js

```diff
@@ -0,0 +1,64 @@
+'use strict'
+
+const isFunction = require('101/is-function')
+const keypather = require('keypather')()
+const Promise = require('bluebird')
+
+const Util = require('./util')
+
+/**
+ * Base class to wrap intercom client.
+ * @class
+ */
+class Base {
+  /**
+   * Constructor for Base
+   * @param {Object} client Intercom client.
+   */
+  constructor (client, name) {
+    this.baseClient = client
+    this.client = keypather.get(client, name)
+    this.name = name
+  }
+
+  /**
+   * Helper method to wrap all client methods.
+   * @param {String} method Name of the method to wrap.
+   * @returns {Promise} Returns success if intercom is disabled, otherwise
+   *   returns results of call to intercom client.
+   * @private
+   */
+  _wrap (method) {
+    return Promise.resolve()
+      .then(() => {
+        if (!Util.canUseIntercom()) { return }
+
+        if (!this.client) {
+          throw new Error(`Base._wrap: invalid ${this.name} client`)
+        }
+
+        const clientMethod = this.client[method]
+        if (!isFunction(clientMethod)) {
+          throw new Error(`
+            Base._wrap: ${this.name} client has no method '${method}'`
```

Comment:
Minor but it looks a bit weird that you're doing `"\`"` on different lines. Stick to all of them in the same line or next to the parentheses. Sad standard doesn't have an opinion on this. Personally, I prefer in the same line since it makes it more readable.